lib/Doctrine/DBAL/Connection.php

```diff
@@ -781,7 +783,7 @@ public function prepare($statement)
             throw DBALException::driverExceptionDuringQuery($this->_driver, $ex, $statement);
         }
 
-        $stmt->setFetchMode($this->defaultFetchMode);
+        call_user_func_array([$stmt, 'setFetchMode'], $this->defaultFetchMode);
```

Comment:
it would be much better to store the information in a way which would not need to rely on ``call_user_func_array``, which is slower than a direct method call